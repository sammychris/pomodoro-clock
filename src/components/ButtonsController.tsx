import React from "react";
import Button from "./Button";

const ButtonsController = ({ title, id, control, increment, decrement }) => {
  return (
    <div id={id}>
      <p>{title}</p>
      <div id={`${id}-label`}>
        <Button
          event={decrement}
          id={`${id}-decrement`}
          iClass="fa fa-arrow-down fa-2x"
        />
        <div id={`${id}-length`}>{control}</div>
        <Button
          event={increment}
          id={`${id}-increment`}
          iClass="fa fa-arrow-up fa-2x"
        />
      </div>
    </div>
  );
};

export default ButtonsController;
