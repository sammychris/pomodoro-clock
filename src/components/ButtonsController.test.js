import { shallow } from "enzyme";
import ButtonsController from "./ButtonsController";

it("expect to render button controller component", () => {
  const wrapper = shallow(<ButtonsController />);
  expect(wrapper.length).toEqual(1);
});

it("expect to render increment button", () => {
  const props = {
    title: "Break",
    id: "break",
    control: jest.fn(),
    increment: jest.fn(),
    decrement: jest.fn(),
  };

  const wrapper = shallow(<ButtonsController {...props} />);
  expect(wrapper.find("#break-increment").length).toEqual(1);
});

it("expect to render decrement button", () => {
  const props = {
    title: "Break",
    id: "break",
    control: jest.fn(),
    increment: jest.fn(),
    decrement: jest.fn(),
  };

  const wrapper = shallow(<ButtonsController {...props} />);
  expect(wrapper.find("#break-decrement").length).toEqual(1);
});

it("expect to take a snap shot of button controller component", () => {
  const wrapper = shallow(<ButtonsController />);
  expect(wrapper).toMatchSnapshot();
});
