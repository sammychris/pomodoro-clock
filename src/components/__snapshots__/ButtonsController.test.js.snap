// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to take a snap shot of button controller component 1`] = `ShallowWrapper {}`;
